src/node_crypto.cc

```diff
@@ -2097,6 +2099,52 @@ void SSLWrap<Base>::GetPeerCertificate(
 }
 
 
+template <class Base>
+void SSLWrap<Base>::GetFinished(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  Base* w;
+  ASSIGN_OR_RETURN_UNWRAP(&w, args.Holder());
+
+  // We cannot just pass nullptr to SSL_get_finished()
+  // because it would further be propagated to memcpy(),
+  // where the standard requirements as described in ISO/IEC 9899:2011
+  // sections 7.21.2.1, 7.21.1.2, and 7.1.4, would be violated.
+  // Thus, we use a dummy byte.
+  char dummy[1];
+  size_t len = SSL_get_finished(w->ssl_, dummy, sizeof dummy);
+  if (!len)
```

Comment:
Style: tiniest of nits but can you use explicit size checks, i.e., `if (len == 0)`?  Likewise on line 2139.

Substance: couldn't you pass in a buffer of size `EVP_MAX_MD_SIZE`?  The result value is (essentially) the HMAC of the message.  You can then drop the `Malloc()` call and switch to `Buffer::Copy()`.

Reply:
> Style: tiniest of nits but can you use explicit size checks, i.e., if (len == 0)? Likewise on line 2139.

Will do.

> Substance: couldn't you pass in a buffer of size `EVP_MAX_MD_SIZE`? The result value is (essentially) the HMAC of the message. You can then drop the `Malloc()` call and switch to `Buffer::Copy()`.

I would prefer avoiding referencing the `EVP_MAX_MD_SIZE` constant if I may.

Reply:
That's fine.  Didn't see until afterwards that you discussed the same thing with @indutny.